spynnaker/pyNN/models/neural_projections/synapse_information.py

```diff
@@ -177,9 +184,29 @@ def may_generate_on_machine(self):
             may have already been so done)
         :rtype: bool
         """
+        # If we are using a virtual machine, we can't generate on the machine
+        if self.__is_virtual_machine:
```

Comment:
no, what im saying is that yopu dont NEED to give synapse_info this virtual machine. theres 2 palces this is called. both of which can figure out if they on a virtual machine and by doing it up in those areas, you reduce a number of loops AND keep low level app specific things app speificic. let the high level things like synapseic_matrixes (which talk sdram afterall) deal with the fiddles of if in a machine.

Reply:
The key here is that those places don't know about the machine, never mind if it is virtual or not.  It would have to be passed down a lot further by moving it out of here.  The place where SynapseInformation is generated already has the information.  Why add it to loads of other calls to get it deep into the synaptic generation process when I can get it there?